### PR TITLE
Correlate merge field data

### PR DESCRIPTION
**Pull request by stevend**
First cut of merging template fields/placeholders with data retrieved from the database query.

Note that currently we are mapping placeholders to db fields in placeholder order. Which means whatever order the placeholders are, its going to pick the corresponding field in the database rows.
This is done starting from the third field in the database row as the first two fields in the database row are the User ID and the User email address.

In future we would potentially change this to allow the user to do the mapping themselves in the UI.

The other change here is to populate the select data tab with the template placeholder fields when a template is selected.

### CODE REVIEW

File: routes.rb [hunk 1]
```diff
@@ -87,7 +95,16 @@ class App < Sinatra::Application
     return {:success => false, :message => I18n.t(:enter_email)}.to_json unless valid_email?(params[:email])
     mandrill = Mandrill.new(session[:key])
     return {:can_connect => false, :message => I18n.t(:mandrill_cannot_connect), :goto_section => 'connect_mandrill'}.to_json unless mandrill.can_connect?
-    response = mandrill.send_single_email(session[:template], params[:email], [{name: 'FULLNAMEORUSERNAME', content: 'King Kong'}])
+
+    begin
+      command = Database.connection.create_command(session[:db_query])
```
Comment (stevend) [inline routes.rb]: not sure if executing the query again is the best idea? Should we cache the query result IF it was executed in the previous step?

File: routes.rb [hunk 2]
```diff
@@ -87,7 +95,16 @@ class App < Sinatra::Application
...
+      data_row = command.execute_reader.take(1).first.values
```
Comment (stevend) [inline routes.rb]: Not sure if executing the query again is the best idea if it was already done in the previous step?

Do we need to cache it if that it the case?

Comment (nocache) [inline routes.rb]: meh, I reckon do it like this. They might not complete the steps in order anyway, right?

**Comment by nocache** [comment]
it didn't seem to toggle to the send test section for me

**Comment by stevend** [comment]
Yep you are right, its because of the change to the db query screen where that code no longer gets executed because of the preview data action that fires on submit.

**Comment by nocache** [comment]
ah I see, does that mean it's my fault? oops

**Comment by stevend** [comment]
Well it happened because of that change but I don't think it was realistic that every tab would only ever have one single action that always triggered a change to another tab.

**Comment by nocache** [comment]
:+1: you can merge this whenever you like

**Comment by stevend** [comment]
I have just put in a temp 'hack' to make the transition work while we ponder better solutions

**Comment by nocache** [comment]
:hammer:

**Pull request merged**